geomesa-accumulo/geomesa-accumulo-spark/src/main/scala/org/locationtech/geomesa/spark/accumulo/AccumuloSpatialRDDProvider.scala

```diff
@@ -26,43 +27,55 @@ import org.locationtech.geomesa.accumulo.index.EmptyPlan
 import org.locationtech.geomesa.index.conf.QueryHints._
 import org.locationtech.geomesa.jobs.GeoMesaConfigurator
 import org.locationtech.geomesa.jobs.accumulo.AccumuloJobUtils
-import org.locationtech.geomesa.jobs.mapreduce.GeoMesaAccumuloInputFormat
+import org.locationtech.geomesa.jobs.mapreduce._
 import org.locationtech.geomesa.spark.SpatialRDDProvider
 import org.locationtech.geomesa.utils.geotools.FeatureUtils
 import org.opengis.feature.simple.SimpleFeature
 import org.opengis.filter.Filter
 
 import scala.collection.JavaConversions._
+import scala.reflect._
 import scala.util.Try
 
 class AccumuloSpatialRDDProvider extends SpatialRDDProvider {
   import org.locationtech.geomesa.spark.CaseInsensitiveMapFix._
+  import org.locationtech.geomesa.spark.TypeDefault._
 
   override def canProcess(params: java.util.Map[String, java.io.Serializable]): Boolean =
     AccumuloDataStoreFactory.canProcess(params)
 
-  def rdd(conf: Configuration,
-          sc: SparkContext,
-          dsParams: Map[String, String],
-          query: Query): RDD[SimpleFeature] = {
-    val ds = DataStoreFinder.getDataStore(dsParams).asInstanceOf[AccumuloDataStore]
-    val username = AccumuloDataStoreParams.userParam.lookUp(dsParams).toString
-    val password = new PasswordToken(AccumuloDataStoreParams.passwordParam.lookUp(dsParams).toString.getBytes)
+  def inputFormat[T : ClassTag](): Class[InputFormat[Text, T]] = {
+    (classTag[T] match {
+      case c if c == classTag[SimpleFeature] => classOf[GeoMesaAccumuloInputFormat]
```

Comment:
to follow up on my earlier comment - instead of this block, have something like:
```
sealed trait SpatialRDDFormat[T]
case object SimpleFeatureFormat extends SpatialRDDFormat[SimpleFeature]
...
def rdd(conf: Configuration, sc: ..., params: ..., query: ...)(implicit format: SpatialRDDFormat[T] = SimpleFeatureFormat): RDD[T]
```